### PR TITLE
refactor: Extract ResourceRadialChart to a separate component

### LINKED ISSUE #3735: Refactor of Resources charts
**Description**
The logic responsible for displaying Radial chart with resources is getting more and more complicated because of unit formatting.
I propose to create component `ResourceRadialChart` which will deal with data format.
Additioanly it will remove duplication and improve readibility of our components.

**Reasons**
Currenlty the data format logic is implemented on normal view which uses directly UI5RadialChart.

**Attachments**
Example code responsible for displaying and formatting Chart:
```
        <UI5RadialChart
          color="var(--sapChart_OrderedColor_5)"
          value={
            cpusToHumanReadable(resources?.requests?.cpu, {
              unit: 'm',
            }).value
          }
          max={
            cpusToHumanReadable(cpu.capacity, {
              unit: 'm',
            }).value
          }
          additionalInfo={`${
            cpusToHumanReadable(resources?.requests?.cpu, {
              unit: 'm',
            }).string
          } / ${
            cpusToHumanReadable(cpu.capacity, {
              unit: 'm',
            }).string
          }`}
        />
```

### PR DESCRIPTION
**Pull request by awoloszyn22**
**Description**

Changes proposed in this pull request:

- Extracted UI5RadialChart with complex display logic to a separate component (ResourceRadialChart) and adjusted its usages

**Related issue(s)**
Resolves #3735

**Definition of done**

- [x] The PR's title starts with one of the following prefixes:
  - feat: A new feature
  - fix: A bug fix
  - docs: Documentation only changes
  - refactor: A code change that neither fixes a bug nor adds a feature
  - test: Adding tests
  - revert: Revert commit
  - chore: Maintainance changes to the build process or auxiliary tools, libraries, workflows, etc.
- [x] Related issues are linked. To link internal trackers, use the issue IDs like `backlog#4567`
- [x] Explain clearly why you created the PR and what changes it introduces
- [x] All necessary steps are delivered, for example, tests, documentation, merging

### CODE REVIEW

**Review by mrCherry97 — CHANGES_REQUESTED** [review]
On cluster overview:

Image

And also I agree with a Claude:
- Wrong default unit for bytes. The unit prop defaults to 'm' (a CPU millicore unit), but ClusterStats.tsx uses valueType="bytes" without passing unit. This means bytesToHumanReadable(memory.usage, { unit: 'm' }) is called, which is semantically incorrect. NodeResources and ResourcesUsage both correctly pass unit="Mi" for memory usage. ClusterStats should too, or the default should be split per valueType.

- Repeated formatting calls. cpusToHumanReadable / bytesToHumanReadable is called 4 times per render (twice for value, twice for max) to build both chart values and the additionalInfo string. Could be called twice and results reused, but this is a minor efficiency concern.

- Implicit bytes fallback. When valueType is undefined, the component silently falls through to bytesToHumanReadable. This isn't documented in the type or prop description, making the API less obvious.

- No tests. The issue doesn't require tests explicitly, but the new formatting logic in ResourceRadialChart is a good candidate for a unit test (especially edge cases like value=0 or max=0).

**Comment by awoloszyn22** [comment]
Follow-up issue for fixing the console warning: https://github.com/kyma-project/busola/issues/5139

File: src/shared/components/ResourceRadialChart/test/ResourceRadialChart.cy.jsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
+/* global cy, describe, it */
+import { ResourceRadialChart } from '../ResourceRadialChart';
+
+describe('ResourceRadialChart', () => {
```
Comment (mrCherry97) [inline src/shared/components/ResourceRadialChart/test/ResourceRadialChart.cy.jsx:4]: Testing the CPU valueType doesn't make sense from my pov. Please add more cases with bytes and different units, maybe.

**Review by mrCherry97 — CHANGES_REQUESTED** [review]

**Review by mrCherry97 — APPROVED** [review]
